wit-0.3.0-draft/system-clock.wit

```diff
@@ -5,20 +5,20 @@ package wasi:clocks@0.3.0;
 /// It is intended to be portable at least between Unix-family platforms and
 /// Windows.
 ///
-/// An "instant", or "exact time", is a point in time without regard to any time
-/// zone: just the time since a particular external reference point, often
-/// called an "epoch".
-///
 /// External references may be reset, so this clock is not necessarily
 /// monotonic, making it unsuitable for measuring elapsed time.
 ///
 /// It is intended for reporting the current date and time for humans.
 @since(version = 0.3.0)
 interface system-clock {
-    /// An exact time in seconds plus nanoseconds.
+    /// An "instant", or "exact time", is a point in time without regard to any
+    /// time zone: just the time since a particular external reference point,
+    /// often called an "epoch".
+    /// For more on various different ways to represent time, see
+    /// https://tc39.es/proposal-temporal/docs/timezone.html
     @since(version = 0.3.0)
     record instant {
-        seconds: u64,
+        seconds: s64,
         nanoseconds: u32,
```

Comment:
This is correct IMO, but might be worth a note that nanoseconds are unsigned because even negative seconds values are still conceptually "positive" exact times.

Reply:
Oh, I was just thinking of this as, the way you'd represent one nanosecond before the epoch would be  { -1 seconds, +999_999_999 nanoseconds }.

I'm not sure what you mean by 'conceptually "positive" exact times'.

Reply:
Agreed, I think of it in the same way as you. What I meant is Dec 31 1969 isn't "negative" in any way other than being before the epoch, which is arbitrary.

The (wrong) alternative interpretation would be to think of { seconds, nanoseconds } as a "duration" before the epoch, e.g. -1.999999999 nanoseconds. That would result in Dec 31 1969 23:59:58.000000001.

Reply:
Added a note which hopefully clarifies:

> Note that even if the seconds field is negative, incrementing nanoseconds always represents moving forwards in time. For example, `{ -1 seconds, 999_999_999 nanoseconds }` represents the instant one nanosecond before the epoch.